rosserial_client/src/ros_lib/ros/msg.h

```diff
@@ -63,7 +63,11 @@ class Msg
    */
   static int serializeAvrFloat64(unsigned char* outbuffer, const float f)
   {
-    const int32_t* val = (int32_t*) &f;
+    // Cast through void* to silence a GCC warning about this being a
+    // non-portable cast (we don't care, it only has to work on AVR).
+    const void* vval = reinterpret_cast<const void*>(&f);
+    const int32_t* val = reinterpret_cast<const int32_t*>(vval);
```

Comment:
Could potentially slip this by the compiler using a `union` as well, but this is nice an explicit. We're doing a bad thing, and we should feel bad.